crypto/rand/rand_unix.c

```diff
@@ -313,25 +370,38 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 
 #   ifdef OPENSSL_RAND_SEED_DEVRANDOM
     bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
-    if (bytes_needed > 0) {
-        static const char *paths[] = { DEVRANDOM, NULL };
-        FILE *fp;
-        int i;
+    {
+        size_t i;
 
-        for (i = 0; paths[i] != NULL; i++) {
-            if ((fp = fopen(paths[i], "rb")) == NULL)
-                continue;
-            setbuf(fp, NULL);
+        for (i = 0; bytes_needed > 0 && i < OSSL_NELEM(devrandom_paths); i++) {
+            FILE *fp = devrandom_files[i];
+
+            if (fp == NULL) {
+                if ((fp = fopen(devrandom_paths[i], "rb")) == NULL)
```

Comment:
Since this is the _unix file, it might be useful to use open() instead of fopen() ...

Reply:
The existing code used fopen(3) and I stayed with that.  I think @richsalz avoided open(2) deliberately.  What do others think?  I'm happy to change it.

There is also an issue with the reading code later -- if less than the full asked for amount is received, it is ignored.  This is perplexing.  Shouldn't whatever is read be used and accounted for?  Unfortunately, the code is sufficiently complex to not be easily understandable.  @mspncp any suggestion here?

Reply:
Considering we don't use any of the more advanced stdio functions, care about line endings and turn off buffering, I'd say that preferring FILE pointers over file descriptors is a moot point, and if using file descriptors gives us an advantage (such as `fstat` mentioned elsewhere), it might as well be changed.  One unnecessary layer removed...

Reply:
As for the bytes read being dropped unless it's all the bytes needed, I'm frankly not sure about that.  Totally spaced on that last time I looked at this.  That could surely be changed, no?  Question is if it's something for this PR...  (at the very least, it should be made a separate commit)

Reply:
> Shouldn't whatever is read be used and accounted for? 

I don't see any reason to discard the random data, just because it's less than the requested amount. Just grab anything you can get, add it to the pool, and drain the next source ;-).

Reply:
You can use fileno() if you stay with FILE * and want to call fstat()